### PR TITLE
chore(QTM-284): Includes correct representation of modal on regression tests

### PR DESCRIPTION
**Pull request by toncabral88**
## Description
https://jirasoftware.catho.com.br/browse/QTM-284

## Setup
to view the components behavior, run `yarn storybook`

## Review guide
- [ ] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [ ] Unit tests (yarn test:components)
- [ ] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review
- [ ] a11y

### Browsers review
- [ ] Layout review
  - [ ] Edge
  - [ ] Firefox
  - [ ] Chrome
  - [ ] Safari
  - [ ] Mobile
- [ ] Ux review validation

### CODE REVIEW

File: .storybook/regression-test/preview.js [hunk 1]
```diff
@@ -4,10 +4,10 @@ import { UiStyle } from '@catho/quantum-storybook-ui';
 import GlobalStyle from '../../components/GlobalStyle';
 
 const CSSDecorator = (Story, context) => (
-  <>
+  <div id="regression-tests">
```
Comment (rapahaeru) [inline .storybook/regression-test/preview.js:7]: how about this name in singular? `regression-test`

File: components/Modal/Modal.jsx [hunk 2]
```diff
@@ -85,8 +88,10 @@ class Modal extends React.Component {
   }
 
   componentDidMount() {
-    const { body } = document;
-    body.appendChild(this.modalOverlay);
+    if (document.querySelector('#regression-tests')) {
```
Comment (rapahaeru) [inline components/Modal/Modal.jsx:91]: this condicional will be necessary in all components that use Portal (SnackBar for example).
How about of export in a method to abstract it.

Could be placed in `components/shared` folder, for example. 

called `regressionTestSelector` for example. WDYT?

**Review by rapahaeru — APPROVED** [review]
## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [x] Code review

**Review by rapahaeru — APPROVED** [review]

**Review by matheusrpp — APPROVED** [review]
## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [x] Code review

### Browsers review
- [ ] Layout review
  - [x] Edge
  - [x] Firefox
  - [x] Chrome
  - [ ] Safari
  - [ ] Mobile

**Pull request merged**